public/components/event_analytics/explorer/events_views/data_grid.tsx

```diff
@@ -3,148 +3,271 @@
-      )
-    );
-  }, [rows, explorerFields.queriedFields]);
-
-  const headers = useMemo(() => getHeaders(explorerFields.selectedFields, DEFAULT_COLUMNS), [
-    explorerFields.selectedFields,
-  ]);
-  const [tableRows, setTableRows] = useState<any[]>([]);
-  useEffect(() => {
-    const dataToRender =
-      explorerFields?.queriedFields && explorerFields.queriedFields.length > 0 ? rowsAll : rows;
-    setTableRows(
-      getTrs(
-        http,
-        explorerFields.selectedFields,
-        limit,
-        setLimit,
-        PAGE_SIZE,
-        timeStampField,
-        explorerFields,
-        pplService,
-        rawQuery,
-        rowRefs,
-        setRowRefs,
-        onFlyoutOpen,
-        dataToRender
-      )
-    );
-  }, [rows, explorerFields.selectedFields]);
-
-  useEffect(() => {
-    setQueriedtableRows((prev) =>
-      getTrs(
-        http,
-        explorerFields.queriedFields,
-        limit,
-        setLimit,
-        PAGE_SIZE,
-        timeStampField,
-        explorerFields,
-        pplService,
-        rawQuery,
-        rowRefs,
-        setRowRefs,
-        onFlyoutOpen,
-        rows,
-        prev
-      )
-    );
-    const dataToRender =
-      explorerFields?.queriedFields && explorerFields.queriedFields.length > 0 ? rowsAll : rows;
-    setTableRows((prev) =>
-      getTrs(
-        http,
-        explorerFields.selectedFields,
-        limit,
-        setLimit,
-        PAGE_SIZE,
-        timeStampField,
-        explorerFields,
-        pplService,
-        rawQuery,
-        rowRefs,
-        setRowRefs,
-        onFlyoutOpen,
-        dataToRender,
-        prev
-      )
-    );
-  }, [limit]);
+  const redoQuery = () => {
+    let finalQuery = '';
+
+    const start = datemath.parse(startTime)?.utc().format(DATE_PICKER_FORMAT);
+    const end = datemath.parse(endTime, { roundUp: true })?.utc().format(DATE_PICKER_FORMAT);
+    const tokens = rawQuery.replaceAll(PPL_NEWLINE_REGEX, '').match(PPL_INDEX_INSERT_POINT_REGEX);
+
+    finalQuery = `${tokens![1]}=${
+      tokens![2]
+    } | where ${timeStampField} >= '${start}' and ${timeStampField} <= '${end}'`;
+
+    finalQuery += tokens![3];
+
+    for (let i = 0; i < sortingFields.current.length; i++) {
+      const field = sortingFields.current[i];
+      const dir = field.direction === 'asc' ? '+' : '-';
+      finalQuery = finalQuery + ` | sort ${dir} ${field.id}`;
+    }
+
+    finalQuery =
+      finalQuery +
+      ` | head ${pageFields.current[1]} from ${pageFields.current[0] * pageFields.current[1]}`;
+    getEvents(finalQuery);
+  };
+
+  // setSort and setPage are used to change the query and send a direct request to get data
+  const setSort = (sort: GridSortingColumn[]) => {
+    sortingFields.current = sort;
+    redoQuery();
+  };
+
+  const setPage = (page: number[]) => {
+    pageFields.current = page;
+    redoQuery();
+  };
+
+  // creates the header for each column listing what that column is
+  const dataGridColumns = useMemo(() => {
```

Comment:
any perf issue needs useMemo? i don't think datagrid itself is memorized, not sure if useMemo/useCallback helps here

Reply:
I was modeling this after Discover's data grid, and they have memoized table attributes. They also have their entire data grid memoized, would it be better if I did that as well? I assumed this was done to stop the rerendering of constant (barring user change) attributes such as columns but would this have no effect without the entire table memoization?

Reply:
it can help if we run into performance issues, but right now useMemo doesn't really help since data grid is not memorized

Reply:
Should I memoize data grid, remove useMemo in my code, or just leave it alone?

Reply:
i think you can remove or leave it if you don't see performance issues right now. we can add optimizations later when needed

Reply:
I'll leave it in, I believe we might need to include some performance optimizations in later